src/aws-lambda/recipes/recipes.py

```diff
@@ -1,13 +1,75 @@
 import json
+import re
+import boto3
+
+dynamodb = boto3.resource('dynamodb')
+ssm = boto3.client('ssm')
+
+response = ssm.get_parameter(Name='retaildemostore-recipes-table-name')
+ddbtable = dynamodb.Table(response['Parameter']['Value'])
+
+def getAllRecipes():
```

Comment:
I was talking to @HieronymusLex about the schema here and if use this pattern we can avoid the full db scan (which would require client side filtering if we add something like tags to the table).

```
#For recipes:
PK: recipe ID
SK: RECIPE (i.e., just the string RECIPE)

# For recipe ingredients
PK: <recipe ID>
SK: INGREDIENT#ingredient_id

# For recipe tags
PK: recipe ID
SK: TAG#tag_name
```
We add a reverse GSI on PK and SK. Then the queries become:

```
All recipes: query the GSI for SK == RECIPE
Recipe by ID, getItem PK == ID, SK == RECIPE
Get recipes by ingredient, query the GSI for SK == INGREDIENT#ID
Get recipes by tag, query the GSI for SK == Tag#tag_name
```